ios/SitumPlugin.m

```diff
@@ -209,6 +209,40 @@ - (dispatch_queue_t)methodQueue
     }];
 }
 
+RCT_EXPORT_METHOD(fetchTilesFromBuilding:(NSDictionary *)buildingJO withSuccessCallback:(RCTResponseSenderBlock)successBlock errorCallback:(RCTResponseSenderBlock)errorBlock)
+{
+    NSString *operation = @"Fetching tiles from building request";
+    if (IS_LOG_ENABLED) {
+        NSLog(@"%@", [NSString stringWithFormat: @"%@ %@ with parameters: %@", DEFAULT_SITUM_LOG, operation, buildingJO]);
+    }
+
+    NSString *buildingId = [buildingJO valueForKey:@"identifier"];
+    
+    [[SITCommunicationManager sharedManager] fetchTilesForBuilding:buildingId success:^(NSDictionary * _Nullable mapping) {
+        if (IS_LOG_ENABLED) {
+            NSLog(@"%@", [NSString stringWithFormat: @"%@ %@ Fetching tiles from building returned values", DEFAULT_SITUM_LOG, operation]);
+        }
+
+        // Parse and convert to json
+        // NSDictionary *buildingInfoJson = [SitumLocationWrapper.shared buildingInfoToJsonObject:mapping[@"results"]];
+
+        // Send result outsidecon e
+        if (IS_LOG_ENABLED) {
+            NSLog(@"%@", [NSString stringWithFormat: @"%@ %@ retrieved tiles from building: %@ with result: %@", DEFAULT_SITUM_LOG, operation, buildingJO, mapping]);
```

Comment:
Este log e practicamente o mesmo que o anterior solo que aquí metes o mapping tamen. 

Nesta función case todo o 70% do código son logs. Entendo que para implementar a funcionalidade pode ter sentido, pero non se pode simplificar un pouco para que quede máis limpo o código final?